### PR TITLE
Modorders 852 some unit tests are not executed

### PR DESCRIPTION
**Pull request by yuntianhu**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/PieceApiTest.java [hunk 1]
```diff
@@ -47,21 +8,33 @@
...
+import static org.folio.TestUtils.getMockAsJson;
+import static org.folio.TestUtils.getMockData;
+import static org.folio.orders.utils.ResourcePathResolver.*;
+import static org.folio.rest.core.exceptions.ErrorCodes.REQUEST_FOUND;
+import static org.folio.rest.impl.MockServer.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
 public class PieceApiTest {
```
Comment (azizbekxm) [inline src/test/java/org/folio/rest/impl/PieceApiTest.java:38]: you should fix your import style

Comment (azizbekxm) [reply src/test/java/org/folio/rest/impl/PieceApiTest.java:38]: please see this guide https://github.com/folio-org/mod-gobi/pull/206#issuecomment-1750115325 (you don't need to download file)

File: src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java [hunk 2]
```diff
@@ -42,32 +44,33 @@ public Future<Void> updateOrCreateTransactionSummary(String orderId, int number,
...
+      .findFirst()
+      .orElse(null);
+
+    // update or create summary if not exists
+    if (CollectionUtils.isEmpty(holder.getEncumbrancesFromStorage())) {
+      return updateOrCreateTransactionSummary(orderId, holder.getAllEncumbrancesQuantity(), requestContext);
+    } else if (holder.getAllEncumbrancesQuantity() == 0) {
+      return Future.succeededFuture();
+    } else {
+      return updateTransactionSummary(orderId, holder.getAllEncumbrancesQuantity(), requestContext);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java:70]: please see this shortcut and reformat particular part of your code not all places

Screenshot 2024-01-22 at 19 08 54

File: src/test/java/org/folio/rest/impl/TitlesApiTest.java [hunk 3]
```diff
@@ -318,7 +318,7 @@ void deleteTitleByIdTest() {
   @Test
   void deleteTitlesByIdWithInvalidFormatTest() {
     logger.info("=== Test delete title by id - bad Id format 400 ===");
-    verifyDeleteResponse(String.format(TITLES_ID_PATH, ID_BAD_FORMAT), TEXT_PLAIN, 400);
+    verifyDeleteResponse(String.format(TITLES_ID_PATH, ID_BAD_FORMAT),  TEXT_PLAIN, 400);
```
Comment (azizbekxm) [inline src/test/java/org/folio/rest/impl/TitlesApiTest.java:321]: extra space

File: src/test/java/org/folio/rest/impl/ReceivingHistoryApiTest.java [hunk 4]
```diff
@@ -155,7 +156,6 @@ void testGetReceivingHistoryForPurchaseOrderWithError() {
   @Test
   void testGetReceivingHistoryBadRequest() {
     logger.info("=== Test Get Receiving History - With Bad Request");
-
```
Comment (azizbekxm) [inline src/test/java/org/folio/rest/impl/ReceivingHistoryApiTest.java:158]: it is better to sperate two line to be clear to reader

File: src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java (hunk 2 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java:70]: and it is better to revert changes relate to reformation of code, only keep important parts

File: src/main/java/org/folio/service/TagService.java [hunk 5]
```diff
@@ -23,7 +23,7 @@ public class TagService {
 
   private static final String TAG_ENDPOINT = resourcesPath(TAGS);
 
-  private final RestClient restClient;
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/TagService.java:26]: this should not be deleted, in scope of your pr you should create/refactor the logic only that connected with your story, not all places in the module

**Comment by azizbekxm** [comment]
image
also please add tests which you have fixed

File: src/test/java/org/folio/rest/impl/PieceApiTest.java (hunk 1 above)
Comment (yuntianhu) [reply src/test/java/org/folio/rest/impl/PieceApiTest.java:38]: will do

File: src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java (hunk 2 above)
Comment (yuntianhu) [reply src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java:70]: sure, will do

File: src/main/java/org/folio/service/TagService.java (hunk 5 above)
Comment (yuntianhu) [reply src/main/java/org/folio/service/TagService.java:26]: Sure, sorry, I will not do it again

File: src/test/java/org/folio/rest/impl/crud/ConfigurationCrudTest.java [hunk 6]
```diff
@@ -1,30 +1,8 @@
...
-import static org.folio.TestConstants.ID_FOR_INTERNAL_SERVER_ERROR;
-import static org.folio.TestConstants.X_ECHO_STATUS;
-import static org.folio.TestConstants.X_OKAPI_USER_ID;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.notNullValue;
-
-import java.util.UUID;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeoutException;
-
```
[the remark this answers is not in the document]
Comment (yuntianhu) [reply src/test/java/org/folio/rest/impl/crud/ConfigurationCrudTest.java:27]: sorry, somehow this test keeps fail on my local machine. I will remember it. and I won't do it again

File: src/test/java/org/folio/rest/impl/TitlesApiTest.java (hunk 3 above)
Comment (yuntianhu) [reply src/test/java/org/folio/rest/impl/TitlesApiTest.java:321]: sure, thank you

**Pull request closed without merging**